src/include/nodes/plannerconfig.h

```diff
@@ -44,16 +44,13 @@ typedef struct PlannerConfig
 
 	bool		honor_order_by;
 
-	/*
-	 * True if it's possible for the plan to depend on external parameters,
-	 * for example if it is a clause in a lateral join or a correlated subquery.
-	 * Doesn't guarantee the subquery is actually correlated
-	 */
-	bool		can_have_dependencies;
+	bool		is_under_subplan; /* True for plan rooted at a subquery which is planned as a subplan */
 
 	/* These ones are tricky */
 	//GpRoleValue	Gp_role; // TODO: this one is tricky
 	bool        force_singleQE; /* True means force gather base rel to singleQE  */
+
+	bool		is_under_lateral; /* True for plan rooted at a lateral join subexpression */
```

Comment:
Why do we need a new boolean field, which also increases the size of the public structure?

Reply:
For readability. We can't rename is_under_subplan, so we either add a new field for this functionality or reuse is_under_subplan while still keeping its name. That is not particularly readable because lateral joins are definitely not subplans, even if we leave a comment about it. Adding a new field seems like a lesser evil.

Reply:
I think, we should not increase size of public structure

Reply:
maybe use macros or union?

Reply:
Macro like what, replacing `is_under_subplan` with `can_have dependencies` everywhere that uses this structure? Is that not too radical for such a small change?
Unions would also break source compatibility, I think. Also, again, they would make this extremely unreadable since the union isn't actually needed by the code.